lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -392,8 +392,8 @@ def _get_azure_cli_profile(self):
         try:
             credentials, subscription_id = get_azure_cli_credentials()
             self._cloud_environment = get_cli_active_cloud()
+            self.azure_credentials = credentials
             return {
```

Comment:
I think perhaps we shouldn't really make this change here. Perhaps we should leave it as it is, and then in constructor:

        elif self.credentials.get('credentials'):
           ....

function name _get_azure_cli_profile() suggests it's just a getter function and it shouldn't really modify anything on its own.

Reply:
Agree.
Also personally i would like to refactor the whole authentication process to be cleaner, because right now it looks a little messy.

Reply:
well, i think right now we should just fix this issue as cleanly as possible. i think we have to learn a bit more before refactoring this file.
Right now i am starting to notice problems with current implementation.